File: java/ServerTerminator.java
class ServerTerminator{
    DoublyLinkedList<Server> queue;
    
    public ServerTerminator(DoublyLinkedList<Server> queue){
        this.queue = queue;
    }
    
    public void terminateNextServer(){
        queue.dequeue().setStatus(Status.TERMINATING);
    }
}

File: cpp/serverTerminator.cpp
#include "serverTerminator.h"
#include "status.h"

ServerTerminator::ServerTerminator(DoublyLinkedList<Server>* queue){
    this->queue = queue;
}

void ServerTerminator::terminateNextServer(){
    this->queue->dequeue()->setStatus(&Status::TERMINATING);
}
